### ISSUE TITLE
"Cannot call New() constructor directly"

### ISSUE THREAD

**Issue by mwolfe02** [WaynePhillipsEA/twinbasic#598]
**Describe the bug**
Class modules in twinBASIC do not allow calling the Class_Initialize() method directly.  VBx allows this.

**To Reproduce**
```
Module MainModule

    ' This project type is set to 'Standard EXE' in the Settings file, so you need a Main() subroutine to run when the EXE is started.
    Public Sub Main()
        Dim Test As New TestClass
        Test.Refresh
    End Sub
	
End Module

Class TestClass
    
    Private Sub Class_Initialize()
        Debug.Print "Initializing..."
    End Sub
    
    Public Sub Refresh()
        Class_Initialize
    End Sub
End Class
```

**Expected behavior**
When this code is run in VBA, it prints "Initializing..." to the Immediate window twice.

**Screenshots**
Behavior in VBA:
![image](https://user-images.githubusercontent.com/305213/147267230-07d2dada-605a-40d6-bac2-e26d7c2181fc.png)

Compile error in tB:
![image](https://user-images.githubusercontent.com/305213/147267481-ba3373cf-53c1-42f6-bd40-fc1cc462558c.png)

**Desktop (please complete the following information):**
 - OS: [Windows 10]
 - twinBASIC compiler version [0.13.55]

**Comment by mansellan** [comment]
While I agree this needs to be replicated for back-compat, it should probably emit a compiler warning!

**Comment by Kr00l** [comment]
Well it may only be allowed to call Class_Initialize within itself... Not the new 'Sub New'. That could be still dissallowed.

**Comment by bclothier** [comment]
> While I agree this needs to be replicated for back-compat, it should probably emit a compiler warning!

Agree. More generally, there should be a warning for any calls into event handlers. Those kind of code are smelly.

**Comment by mansellan** [comment]
@mwolfe02 A better pattern (which I've used before) would be to forward `Class_Initialize` to a `Private Sub`, which could then be also called by other sites. I've used that one myself a few times, and it would also avoid the compiler errors (later, warnings) of calling an event handler directly.

```vb
Module MainModule

    ' This project type is set to 'Standard EXE' in the Settings file, so you need a Main() subroutine to run when the EXE is started.
    Public Sub Main()
        Dim Test As New TestClass
        Test.Refresh
    End Sub
	
End Module

Class TestClass
    
    Private Sub Class_Initialize()
        Refresh
    End Sub
    
    Public Sub Refresh()
        Debug.Print "Initializing..."
    End Sub
End Class
```

**Comment by WaynePhillipsEA** [comment]
Fixed in v0.13.56, thanks for reporting!

**Comment by mwolfe02** [comment]
@mansellan Agreed.  That is a better (and cleaner) pattern.
